app/src/main/java/com/example/campusguide/calendar/Events.kt

```diff
@@ -0,0 +1,139 @@
+package com.example.campusguide.calendar
+
+import android.annotation.SuppressLint
+import android.database.Cursor
+import android.net.Uri
+import android.provider.CalendarContract
+import com.example.campusguide.Constants
+import com.example.campusguide.MapsActivity
+import java.util.*
+import kotlin.collections.ArrayList
+
+class Events constructor(
+    val activity: MapsActivity,
+    private var selectedCalendar: Pair<Long, String>
+) {
+
+    private val contentResolver = activity.applicationContext.contentResolver
+
+    private val EVENT_VALUES_TO_QUERY: Array<String> = arrayOf(
+        CalendarContract.Events.CALENDAR_ID, // long
+        CalendarContract.Events.DTSTART, // start in UTC milliseconds since the epoch
+        CalendarContract.Events.EVENT_LOCATION // string
+    )
+    private val PROJECTION_EVENT_CALENDAR_INDEX: Int = 0
+    private val PROJECTION_EVENT_START_TIME_INDEX: Int = 1
+    private val PROJECTION_EVENT_LOCATION_INDEX: Int = 2
+
+    private val eventUri: Uri = CalendarContract.Events.CONTENT_URI
+    private val todaysEvents: ArrayList<Pair<Long, String>> =
+        arrayListOf() // event time & location str
+
+    // TODO: WIP for UC-16
+    @SuppressLint("MissingPermission")
```

Comment:
Again, why are we suppressing this?

Reply:
addressed in inline comment but i'll respond to you here as well: the permissions are checked elsewhere. the suppression is purely so that androidstudio will stop harassing us about it in every class that uses the permissions we've already checked for